### PR TITLE
parse object description from structure annotation.

### PR DESCRIPTION
**Pull request by zhy1**
**Describe the PR**
parse object description from structure annotation. 

some entity.go
```go

type Post struct {
	ID int `json:"id" example:"1" format:"int64" description:"Post primary field."`
	// Post name
	Name string `json:"name" example:"poti"`
	// Post data
	Data struct {
		// Post tag
		Tag []string `json:"name"`
	} `json:"data"`
}
```

output
swagger.json
```json
 "id": {
                    "description": "Post primary field.",
                    "type": "integer",
                    "format": "int64",
                    "example": 1
                },
```

**Additional context**
if attribute not setting. use the previous rules

### CODE REVIEW

**Comment by ubogdan** [comment]
Before doing any code review.
  Why do you think decorating struct with another tag is more fine then a annotation/comment ?

**Comment by zhy1** [comment]
I generated structure from databases.
put the table filed comment of the database into the structure description.
i need swag to show  it.

**Comment by zhy1** [comment]
```go
Longitude *[]float64 `gorm:"column:longitude;type:Longitude;description:" json:"longitude" format:"[]float64"  description:"last connect longitude, don't change" `
```

**Comment by easonlin404** [comment]
How about your thought for this PR? @pei0804 @ubogdan

File: parser.go [hunk 1]
```diff
@@ -895,19 +903,20 @@ func (parser *Parser) parseStruct(pkgName string, field *ast.Field) (map[string]
 			properties[structField.name] = spec.Schema{
 				SchemaProps: spec.SchemaProps{
 					Type:        []string{structField.schemaType},
-					Description: desc,
+					Description: selectValidItem(structField.description, desc),
```
Comment (ubogdan) [inline parser.go]: Please invert the priority:
comment , struct.field

File: parser.go [hunk 2]
```diff
@@ -860,7 +868,7 @@ func (parser *Parser) parseStruct(pkgName string, field *ast.Field) (map[string]
 		properties[structField.name] = spec.Schema{
 			SchemaProps: spec.SchemaProps{
 				Type:        []string{"object"}, // to avoid swagger validation error
-				Description: desc,
+				Description: selectValidItem(structField.description, desc),
```
Comment (ubogdan) [inline parser.go]: Please invert the priority:
comment , struct.field

File: parser.go [hunk 3]
```diff
@@ -874,7 +882,7 @@ func (parser *Parser) parseStruct(pkgName string, field *ast.Field) (map[string]
 			properties[structField.name] = spec.Schema{
 				SchemaProps: spec.SchemaProps{
 					Type:        []string{structField.schemaType},
-					Description: desc,
+					Description: selectValidItem(structField.description, desc),
```
Comment (ubogdan) [inline parser.go]: Please invert the priority:
comment , struct.field

File: parser.go [hunk 4]
```diff
@@ -925,7 +934,7 @@ func (parser *Parser) parseStruct(pkgName string, field *ast.Field) (map[string]
 		properties[structField.name] = spec.Schema{
 			SchemaProps: spec.SchemaProps{
 				Type:        []string{structField.schemaType},
-				Description: desc,
+				Description: selectValidItem(structField.description, desc),
```
Comment (ubogdan) [inline parser.go]: Please invert the priority:
comment , struct.field

File: parser.go [hunk 5]
```diff
@@ -964,7 +973,7 @@ func (parser *Parser) parseStruct(pkgName string, field *ast.Field) (map[string]
 			properties[structField.name] = spec.Schema{
 				SchemaProps: spec.SchemaProps{
 					Type:        []string{structField.schemaType},
-					Description: desc,
+					Description: selectValidItem(structField.description, desc),
```
Comment (ubogdan) [inline parser.go]: Please invert the priority:
comment , struct.field

**Comment by ubogdan** [comment]
@easonlin404 this PR does not come from an issue. 

My personal trough: tag decoration is ugly and it does not justify if it is already implemented in annotation and is not even used for something else like "validation" by example.

Anyway ,default swagger behavior should be keept (comments from anotations should have priority).

**Comment by easonlin404** [comment]
> @easonlin404 this PR does not come from an issue.
> 
> My personal trough: tag decoration is ugly and it does not justify if it is already implemented in annotation and is not even used for something else like "validation" by example.
> 
> Anyway ,default swagger behavior should be keept (comments from anotations should have priority).

I aggre with you. Maybe we could close this PR.

**Pull request closed without merging**